From 9299cc2194f531634cc8f1ed884baa89eec3a98b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 07 Apr 2020 19:46:59 +0300
Subject: [PATCH] [SW-11180] preboot: boot_desc: fix update pending

Commit:
f5acb4d ("[SW-11180] preboot: boot_desc: update pending reg")
added updating the "updated pending register", but forgot to take into
an account the actual updated pending status.
This resulted in setting the "update pending register" each boot and
executing the U-Boot "post reset" command.
The "update pending register" must be updated only in case there was
actually an update pending... Conform to this.

Fixes: f5acb4d119fe24ac7c9e2fca191ad913e789727f

Change-Id: Ib58b87a4f43cffc728c105a3cad75a9e1fc91f5c
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/preboot.c b/app/preboot.c
index 3d4bf1f..46aa25e 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -97,7 +97,9 @@
 					      HL_FW_IMG_UBOOT, &idx);
 		if (offset != HL_FW_IMG_NOT_FOUND) {
 			*pending = image_pending;
-			preboot_set_upd_pending_reg(idx);
+			if (image_pending)
+				preboot_set_upd_pending_reg(idx);
+
 			return offset;
 		}
 	}
